component/espejote-templates/netpol-sync.jsonnet

```diff
@@ -1,106 +1,170 @@
-        namespace: namespace.metadata.name,
-      },
-    }),
-    esp.markForDelete({
-      apiVersion: resources.allowFromClusterNodes.apiVersion,
-      kind: resources.allowFromClusterNodes.kind,
-      metadata: {
-        name: resources.allowFromClusterNodes.metadata.name,
-        namespace: namespace.metadata.name,
-      },
-    }),
-  ];
+  local defaultPolicySets =
+    if objHasLabel(namespace, config.namespaceLabels.baseDefaults) then
+      [ 'base' ]
+    else
+      [ 'base', 'default' ];
 
-local applyBasePolicies(namespace) = [
-  resources.allowFromOtherWithSpec {
-    metadata+: {
-      namespace: namespace.metadata.name,
-    },
-  },
-] + if config.networkPlugin == 'cilium' then [
-  resources.allowFromNodesWithSpec {
+  // Policy sets based on labels starting with params.labels.policySetPrefix.
+  //   labels:
+  //     set.example.io/airlock: ""
+  //     set.example.io/myapp: ""
+  // would return the policy sets `["airlock", "myapp"]`.
+  // The configured prefix is suffixed with a '/' if it does not already end with one.
+  local policySetsFromLabel =
+    local prefix = if std.endsWith(config.namespaceLabels.policySetPrefix, '/') then
+      config.namespaceLabels.policySetPrefix
+    else
+      config.namespaceLabels.policySetPrefix + '/';
+
+    [
+      lbl[std.length(prefix):]
+      for lbl in std.objectFields(std.get(namespace.metadata, 'labels', {}))
+      if std.startsWith(lbl, prefix) && (namespace.metadata.labels[lbl] == 'true' || namespace.metadata.labels[lbl] == '')
+    ];
+
+  if std.member(config.ignoredNamespaces, namespace.metadata.name) || objHasLabel(namespace, config.namespaceLabels.noDefaults) then
+    []
+  else if std.length(policySetsFromLabel) > 0 then
+    std.set([ 'base' ] + policySetsFromLabel)
+  else
+    std.set(defaultPolicySets);
+
+local isCiliumPolicy(policyName) =
+  std.startsWith(policyName, 'cilium/');
+
+local cniMatches(policyName) =
+  if isCiliumPolicy(policyName) then
+    config.hasCilium
+  else
+    true;
+
+// Generate policy sets.
+local generatePolicyMetadata(policyName, namespace) =
+  (if isCiliumPolicy(policyName) then {
+     apiVersion: 'cilium.io/v2',
+     kind: 'CiliumNetworkPolicy',
+     metadata+: {
+       name: std.strReplace(policyName, 'cilium/', ''),
+     },
+   } else {
+     apiVersion: 'networking.k8s.io/v1',
+     kind: 'NetworkPolicy',
+     metadata: {
+       name: policyName,
+     },
+   }) + {
     metadata+: {
+      annotations: config.netpolAnnotations,
+      labels: config.netpolLabels,
       namespace: namespace.metadata.name,
     },
-  },
-] else [];
-
-local applyDefaultPolicies(namespace) =
-  applyBasePolicies(namespace) + [
-    resources.allowFromSameWithSpec {
-      metadata+: {
-        namespace: namespace.metadata.name,
-      },
-    },
-  ];
+  };
+
+local generatePolicy(policyName, namespace) =
+  generatePolicyMetadata(policyName, namespace) {
+    spec: config.policies[policyName],
```

Comment:
Do I see this right that an unknown policy set is a hard template error right now? Might make more sense to skip unknown sets and emit an event in the target namespace (if we can do this now)?